### PR TITLE
Fix for providing the ability to set 2 subscriptions with the same name

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

File: lib/commands/account._js [hunk 1]
```diff
@@ -61,12 +61,18 @@ exports.init = function (cli) {
       });
     });
 
-  account.command('show [subscription]')
+  account.command('show [subscriptionNameOrId]')
     .description($('Show details about a subscription'))
-    .option('-s --subscription <subscription>', $('The subscription to show'))
+    .option('-s --subscriptionNameOrId <subscriptionNameOrId>', $('The subscription name to show'))
```
Comment (amarzavery) [inline lib/commands/account._js]: remove name

File: lib/commands/account._js [hunk 2]
```diff
@@ -92,19 +93,36 @@ exports.init = function (cli) {
         [$('Registered ARM Namespaces'), 'registeredResourceNamespaces']
       ];
 
-      log.report(format.concat(options.details ? detailsFormat : []), sub);
+      subscriptions.forEach(function (sub) {
+        log.report(format.concat(options.details ? detailsFormat : []), sub);
+        log.data($(''));
+      });
+
```
Comment (yugangw-msft) [inline lib/commands/account._js]: not, remove the empty line

File: lib/util/profile/profile.js [hunk 3]
```diff
@@ -105,26 +105,26 @@ _.extend(Profile.prototype, {
...
-    this.subscriptions[subscription.name] = subscription;
+    this.subscriptions[subscription.id] = subscription;
     subscription.on('updated', this.onSubscriptionUpdated);
   },
 
-  deleteSubscription: function (subscriptionOrName) {
-    var subscription = subscriptionOrName;
-    if (_.isString(subscriptionOrName)) {
-      subscription = this.subscriptions[subscriptionOrName];
+  deleteSubscription: function (subscriptionOrId) {
```
Comment (yugangw-msft) [inline lib/util/profile/profile.js]: A better name would be: subscriptionNameOrID?

**Comment by yugangw-msft** [comment]
lgtm

**Pull request merged**